cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/ParameterJsonStructureTest.java

```diff
@@ -0,0 +1,141 @@
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Set;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.TreeSet;
+
+public class ParameterJsonStructureTest {
+  private Map<String, Class> _endpointToClass;
+  OpenAPI _openAPI;
+  /**
+   * Specify endpoints to be tested
+   */
+  @Before
+  public void setupParameterClasses() {
+    _endpointToClass = new HashMap<>();
+    _endpointToClass.put("/kafkacruisecontrol/rebalance", RebalanceParameters.class);
+    _endpointToClass.put("/kafkacruisecontrol/partition_load", PartitionLoadParameters.class);
+  }
+
+  /**
+   * Load the OpenAPI files for endpoints and compares them against the source code
+   */
+  @Test
+  public void loadOpenApiSpec() throws Exception {
+    OpenAPIV3Parser openApiParser = new OpenAPIV3Parser();
+    String baseFileName = "../schemas/base.yaml";
+    ParseOptions options = new ParseOptions();
+    options.setResolve(true);
+    options.setFlatten(true);
+
+    SwaggerParseResult parseResult = openApiParser.readLocation(baseFileName, null, options);
+    _openAPI = parseResult.getOpenAPI();
+    Schema schema = parseSchema();
+    for (Map.Entry<String, Set<String>> endpoint: schema.getEndpointsAndParameters()) {
+      Assert.assertTrue(_endpointToClass.containsKey(endpoint.getKey()));
+      CruiseControlParameters endpointParams = (CruiseControlParameters) (_endpointToClass.get(endpoint.getKey()).newInstance());
+      Assert.assertEquals(
+        endpoint.getValue(),
+        endpointParams.caseInsensitiveParameterNames());
+    }
+  }
+
+  /**
+   * Return the list of parameters given the path item (an endpoint)
+   *
+   * @param pathItem Endpoint defined as a PathItem object
+   * @return set of parameters for the specified endpoint
+   */
+  public static Set<String> parseEndpoint(PathItem pathItem) throws IllegalArgumentException {
+    List<Parameter> parameterList;
+    Set<String> parameterSet = new TreeSet<>();
+    if (pathItem.getGet() != null) {
+      parameterList = pathItem.getGet().getParameters();
+    } else if (pathItem.getPost() != null) {
+      parameterList = pathItem.getPost().getParameters();
+    } else {
+      throw new IllegalArgumentException("Schema Parser does not support HTTP methods other than GET/POST");
+    }
+
+    for (Parameter parameter : parameterList) {
+      parameterSet.add(parameter.getName());
+    }
+    return parameterSet;
+  }
+
+  /**
+   * parse the entire schema defined in yaml file and return the schema object
+   *
+   * @return object representing the schema defined in the yaml file
+   */
+  public Schema parseSchema() {
+    Schema parsedSchema = new Schema();
+    Paths paths = _openAPI.getPaths();
+    for (Map.Entry<String, PathItem> endpoint : paths.entrySet()) {
+      parsedSchema.addEndpoint(endpoint.getKey(), parseEndpoint(endpoint.getValue()));
+    }
+    return parsedSchema;
+  }
+
+  static class Schema {
```

Comment:
Is there any reason that we do not directly use a `Map<String, Set<String>>` here( and define a specific inner static class `Schema`)?